### PR TITLE
Prevent duplicate title when editing/creating questions

### LINKED ISSUE #45: Prevent users from editing/creating duplicate titles

### PR DESCRIPTION
**Pull request by anthonynmh**
Closes #45 .

### CODE REVIEW

**Review by cadencjk — APPROVED** [review]
LGTM!

File: backend/server/services/questionService.js
Comment (lhy-hoyin) [inline backend/server/services/questionService.js:1]: It looks like the code is duplicated at 2 parts of the same file. Perhaps you can extract this out into a function.
```
const DuplicateTitleTest = () => {
  const isDuplicateTitle = await questionRepository.findByTitle(title);
   if (isDuplicateTitle) {
     throw { status: 409, message: 'Question title already exist' };
   }
};
```

Replace the changed code sections with calling the test:
```
DuplicateTitleTest(); // I think its called like this; play around with it.
```

File: frontend/src/components/EditQuestion.js [hunk 1]
```diff
@@ -66,6 +66,8 @@ const EditQuestion = () => {
     } catch (error) {
       if (error.response.status === 400) {
         showValidationErrorToast(error);
+      } else if (error.response.status === 409) {
+        showDuplicateQuestionErrorToast(error);
       } else {
         showServerErrorToast(error);
```
Comment (lhy-hoyin) [inline frontend/src/components/EditQuestion.js:72]: Would a switch case be better/cleaner than multiple if-else?
```
switch (error.response.status) {
  case 400: showValidationErrorToast(error);        break;
  case 409: showDuplicateQuestionErrorToast(error); break;
  default:  showServerErrorToast(error);
}
```

Comment (anthonynmh) [reply frontend/src/components/EditQuestion.js]: Updated code with switch-case

File: backend/server/services/questionService.js
Comment (anthonynmh) [reply backend/server/services/questionService.js:1]: Extracted

**Pull request merged**